pubsub/drivertest/drivertest.go

```diff
@@ -374,15 +377,32 @@ func testCancelSendReceive(t *testing.T, newHarness HarnessMaker) {
 	}
 	defer cleanup()
 
+	canceledCtx, cancel := context.WithCancel(ctx)
 	cancel()
 
 	m := &pubsub.Message{}
-	if err := top.Send(ctx, m); !isCanceled(err) {
+	if err := top.Send(canceledCtx, m); !isCanceled(err) {
 		t.Errorf("top.Send returned %v (%T), want context.Canceled", err, err)
 	}
-	if _, err := sub.Receive(ctx); !isCanceled(err) {
+	if _, err := sub.Receive(canceledCtx); !isCanceled(err) {
 		t.Errorf("sub.Receive returned %v (%T), want context.Canceled", err, err)
 	}
+
+	// Repeat with a sub.Receive that's interrupted.
+	delayedCancelCtx, cancel := context.WithCancel(ctx)
+	ch := make(chan struct{})
+	go func() {
+		if _, err := sub.Receive(delayedCancelCtx); !isCanceled(err) {
+			t.Errorf("sub.Receive returned %v (%T), want context.Canceled", err, err)
+		}
+		close(ch)
+	}()
+	// Give the goroutine some time to spin up and be blocking in Receive.
+	// There's no guarantee it will get there, but tests will at least be flaky
+	// if it gets there sometimes.
+	time.Sleep(50 * time.Millisecond)
```

Comment:
Maybe this should be longer. 50ms is nothing to Windows. (If the sleep is so short that the delay always happens before Receive blocks, then this test won't detect anything.)

Reply:
Upped to 100ms. I don't think it matters too much though; as the previous comment observes, the test will be flaky if there's a bug and it gets to `Receive` sometimes.